public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -48,12 +132,28 @@ const DimensionFields: React.FC<DimensionFieldsProps> = ({ query, dimensionOptio
               menuShouldPortal
               placeholder="Field"
               value={filter.dimension}
-              options={dimensionOptions}
+              options={getValidDimensionOptions(filter.dimension)}
               onChange={(v) => onFieldChange(index, 'dimension', v.value ?? '')}
               width={38}
             />
-            <InlineLabel aria-label="equals">==</InlineLabel>
-            <Input placeholder="" value={filter.filter} onChange={(ev) => onFilterInputChange(index, ev)} />
+            <Select
+              menuShouldPortal
+              placeholder="Operation"
+              value={filter.operator}
+              options={getValidOperators(filter.operator)}
+              onChange={(v) => onFieldChange(index, 'operator', v.value ?? '')}
+              allowCustomValue
+            />
+            <Select
+              menuShouldPortal
+              placeholder="Select value"
+              value={filter.filter ? filter.filter : ''}
+              allowCustomValue
+              options={getValidFilterOptions(filter.filter, filter.dimension)}
+              onChange={(v) => onFilterInputChange(index, v)}
+              isClearable
```

Comment:
Should `isClearable` allow you to completely clear the select? It wasn't working for me. Probably doing something wrong, but if not this is an issue in grafana ui so nothing to care about in this pr.,